gunicorn/arbiter.py

```diff
@@ -460,10 +461,12 @@ def reap_workers(self):
                     if exitcode == self.APP_LOAD_ERROR:
                         reason = "App failed to load."
                         raise HaltServer(reason, self.APP_LOAD_ERROR)
-                    worker = self.WORKERS.pop(wpid, None)
-                    if not worker:
-                        continue
-                    worker.tmp.close()
+
+                    worker = (self.WORKERS.pop(wpid, None)
```

Comment:
That means that `SIGCHILD` can be caused either by the child exiting unexpectedly, or because we killed it intentionally. In the first case, that worker would have still been in `self.WORKERS`. In the second, we would have moved it to `self.DYING_WORKERS`.